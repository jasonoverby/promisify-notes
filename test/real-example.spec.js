const { get } = require('https');
const { getHttps } = require('../lib/get-https');
const {
  getTodo,
  getTodos,
} = require('../lib/real-example');

describe('real example', () => {
  describe('https.get with callback', () => {
    it('succeeds when url is valid', (done) => {

      const todosUrl = 'https://jsonplaceholder.typicode.com/todos';
      const todoId = 2;
      const callback = (response) => {
        let str = '';
        response.setEncoding('utf8');
        response.on('data', (data) => {
          str += data;
        });
        response.on('end', () => {
          const dataObj = JSON.parse(str);
          expect(dataObj).toMatchSnapshot({
            userId: expect.any(Number),
            id: expect.any(Number),
            title: expect.any(String),
            completed: expect.any(Boolean),
          });
          done();
        });

      };

      get(`${todosUrl}/${todoId}`, callback);
      expect.assertions(1);
    });

    // it('fails when something is not "something"', (done) => {
    //   const something = '';
    //   const theCallback = (result, err) => {
    //     expect(result).toBe(null);
    //     expect(err).toBeInstanceOf(TypeError);
    //     done();
    //   };

    //   myFuncWithCallback(something, theCallback);
    //   expect.assertions(2);
    // });
  });

});
