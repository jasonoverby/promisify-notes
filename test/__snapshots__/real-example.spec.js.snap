// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`real example https.get with callback succeeds when url is valid 1`] = `
Object {
  "completed": Any<Boolean>,
  "id": Any<Number>,
  "title": Any<String>,
  "userId": Any<Number>,
}
`;
